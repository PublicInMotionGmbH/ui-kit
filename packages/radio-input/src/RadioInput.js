import React from 'react'
import PropTypes from 'prop-types'

import { buildClassName } from '@talixo/shared'

const propTypes = {
  /** Radio button description.  */
  children: PropTypes.node,

  /** Additional wrapper class name. */
  className: PropTypes.string,

  /** Indicates if option should be disabled. */
  disabled: PropTypes.bool,

  /** Radio input label size ('small', 'large'). */
  size: PropTypes.oneOf([ 'small', 'large' ]),

  /** Styles passed to radio button wrapper. */
  style: PropTypes.object
}

const defaultProps = {
  disabled: false
}

/**
 * Component which represents Radio Input.
 *
 * @param {object} props
 * @param {*} [props.children]
 * @param {string} [props.className]
 * @param {string} [props.size]
 * @param {object} [props.style]
 * @returns {React.Element}
 */
function RadioInput (props) {
  const { children, className, disabled, size, style, onChange, ...passedProps } = props

  const wrapperClass = buildClassName('radio-input', className, [ size ], { disabled })
  const inputClass = buildClassName(['radio-input', 'input'])

  const handleOnChange = e => {
    if (onChange) { onChange(e.target.checked, e) }
  }

  return (
    <label className={wrapperClass} style={style}>
      <input
        className={inputClass}
        disabled={disabled}
        type='radio'
        onChange={handleOnChange}
        {...passedProps}
      />
      <span>{children}</span>
    </label>
  )
}

RadioInput.propTypes = propTypes

RadioInput.defaultProps = defaultProps

export default RadioInput
