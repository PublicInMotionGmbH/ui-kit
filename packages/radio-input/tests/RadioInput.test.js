import React from 'react'
import RadioInput from '../src/RadioInput'
import { shallow } from 'enzyme'

describe('<RadioInput />', () => {
  it('renders children correctly', () => {
    const wrapper = shallow(<RadioInput />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders name correctly', () => {
    const wrapper = shallow(<RadioInput name='gender'>Radio label</RadioInput>)
    expect(wrapper).toMatchSnapshot()
  })
})
