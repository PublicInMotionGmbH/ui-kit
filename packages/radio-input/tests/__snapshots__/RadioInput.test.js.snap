// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioInput /> renders children correctly 1`] = `
<label
  className="talixo-radio-input"
>
  <input
    className="talixo-radio-input__input"
    disabled={false}
    onChange={[Function]}
    type="radio"
  />
  <span />
</label>
`;
